### PR TITLE
Closes #343: Write the whole mechanizm of aff matching

### LINKED ISSUE #343: Write the whole mechanizm of aff matching
- AffMatcher
- AffOrgJoiner
- AffOrgComparator
- Model for results with wrting it to files
- Test

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/bucket/StringPartFirstLettersHasherTest.java [hunk 1]
```diff
@@ -51,7 +52,7 @@ public void hash_value_null() {
         
         // execute & assert
         
-        assertEquals("", hasher.hash(" ", 1, 1));
+        assertEquals("", hasher.hash(" "));
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/bucket/StringPartFirstLettersHasherTest.java]: this test is identical to `hash_value_blank()`

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/AffOrgMatchComputer.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        
+    }
+    
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    /**
+     * Computes the match strength of every passed pair of affiliation and organization. The returned strengths are in the range
+     * of (0, 1>. The pairs which have the match strength equal to 0 are filtered out and not returned.<br/><br/>
```
Comment (madryk) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/AffOrgMatchComputer.java]: Standard notation for interval is `(0, 1]`

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/bucket/StringPartFirstLettersHasherTest.java (hunk 1 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/bucket/StringPartFirstLettersHasherTest.java]: :+1:

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/AffOrgMatchComputer.java (hunk 2 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/match/AffOrgMatchComputer.java]: ok

**Pull request closed without merging**